src/Sarif.Viewer.VisualStudio/Controls/InfoBar.cs

```diff
@@ -197,5 +217,21 @@ public void OnActionItemClicked(IVsInfoBarUIElement infoBarUIElement, IVsInfoBar
                 this.clickAction?.Invoke(actionItem);
             }
         }
+
+        private static async Task AddInfoBarIfRequiredAsync(
+            ExceptionalConditions detectedConditions,
+            ExceptionalConditions individualCondition,
+            string message,
+            ImageMoniker imageMoniker)
+        {
+            // It's safe to access and update the dictionary because this method is called inside
+            // the write lock.
+            if ((detectedConditions & individualCondition) == individualCondition && !s_infoBarToConditionDictionary.Values.Contains(individualCondition))
```

Comment:
Why is it not just "detectedConditions == individualCondition "? In c#,  enum type values can be comparied with "==".

Reply:
Because `detectedConditions` is a `Flags` enum -- the same log can exhibit multiple exceptional conditions. I need to know if this particular bit is turned on, regardless of what other bits might be turned one.

---
In reply to: [527204867](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527204867) [](ancestors = 527204867)